src/test/java/teammates/test/cases/browsertests/StudentProfilePageUiTest.java

```diff
@@ -297,6 +297,23 @@ private void testAjaxPictureUrl() {
         return loginAdminToPage(profileUrl, typeOfPage);
     }
 
+    private <T extends AppPage> T loginAsInstructorToPage(String instructorId, AppUrl url, Class<T> typeOfPage) {
+        //logout
+        logout();
+
+        //login
+        getHomePage().clickInstructorLogin();
+
+        //login based on the login page type
+        LoginPage loginPage = AppPage.createCorrectLoginPageType(browser);
+        loginPage.loginAsInstructor(instructorId, TestProperties.TEST_INSTRUCTOR_PASSWORD);
```

Comment:
I understand that using the instructor home page was to avoid an `InvocationTargetException` at this step, as `LoginPage#loginAsInstructor` expects an instructor home page. However, I think the right way to do things would be to overload `loginAsInstructor` with a method that accepts any page type, just like the existing `LoginPage#loginAsStudent`.

Reply:
Isn't it out of the scope of this PR, changing the behaviour of the `LoginPage`?

Reply:
Also, `LoginPage#loginAsStudent` also expects `StudentHomePage`?

Reply:
Actually, I think there will not be any exception, since the return value is never used. Let me try and see how it goes.

Reply:
There's another method: https://github.com/TEAMMATES/teammates/blob/19a0e8ee0e7d6a9e609de972257060fdee2d15f9/src/test/java/teammates/test/pageobjects/LoginPage.java#L15